connectors/steps/snowflake/connect.py

```diff
@@ -0,0 +1,43 @@
+import snowflake.connector
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+
+def connect(
+    username: str,
+    password: str,
+    private_key: str,
+    passphrase: str,
+    account: str,
+    warehouse: str,
+    database: str,
+    schema: str,
+):
+    if private_key:
+        key = f"-----BEGIN ENCRYPTED PRIVATE KEY-----\n{private_key}\n-----END ENCRYPTED PRIVATE KEY-----"
```

Comment:
How about decouple the decryptic logic? 

def decrypt_key(private_key: str, passphrase: str):
    key = f"-----BEGIN ENCRYPTED PRIVATE KEY-----\n{private_key}\n-----END ENCRYPTED PRIVATE KEY-----"
    p_key = serialization.load_pem_private_key(
        key.encode(),
        password=passphrase.encode(),
        backend=default_backend(),
    )
    pkb = p_key.private_bytes(
        encoding=serialization.Encoding.DER,
        format=serialization.PrivateFormat.PKCS8,
        encryption_algorithm=serialization.NoEncryption(),
    )
    return pkb